multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2019,2021 Precisely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.precisely.bigdata.sample.spark
+
+import com.pb.downloadmanager.api.DownloadManagerBuilder
+import com.pb.downloadmanager.api.downloaders.LocalFilePassthroughDownloader
+import com.pb.downloadmanager.api.downloaders.hadoop.{HDFSDownloader, S3Downloader}
+import com.precisely.addressing.v1.model.RequestAddress
+import com.precisely.bigdata.addressing.spark.api.{AddressingBuilder, AddressingExecutor, RequestInput}
+import com.precisely.addressing.v1.model.PreferencesBuilder
+import org.apache.commons.collections.CollectionUtils
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.{DataFrame, SaveMode, SparkSession}
+import org.apache.spark.sql.functions.{callUDF, col, lit, map}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import java.util.Arrays;
+
+object MultipassAddressing {
+  def main(args: Array[String]): Unit = {
+    val inputAddressPath = args(0)
+    val resourcesLocation = args(1)
+    val dataLocation = args(2)
+    val downloadLocation = args(3)
+    val outputDirectory = args(4)
+
+    val sparkConf = new SparkConf()
+    sparkConf.setIfMissing("spark.master", "local[*]")
+    val session = SparkSession.builder()
+      .config(sparkConf)
+      .getOrCreate()
+
+    // Load the addresses from the csv
+    val addressInputDF = AddressInput.open(session, inputAddressPath)
+
+    // set preference to return all fields
+    val prefBuilder = new PreferencesBuilder().withReturnAllInfo(true).build()
```

Comment:
This variable doesn't hold a builder, it should be named preferences instead.

Reply:
Ok